spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -38,6 +38,13 @@
  */
 public interface SpannerOperations {
 
+	/**
+	 * Gets the key for the given object.
+	 * @param object
+	 * @return
+	 */
+	Key getId(Object object);
```

Comment:
Typically you would not expose a `getKey(…)` method but rather handle this aspect internally. It's up to the Template API to figure out a key. That said, `findById(…)` methods accept typically `Object` (`findById(Object, Class)`).

Reply:
Our other find() methods do take the Key type, and a Class, so I thought it made sense to provide the the user a way to get a Key. Since the Key type was already exposed to them I think this isn't showing them something they otherwise wouldn't need to know.

Reply:
The consequence of not exposing `Key` would be to accept `Object` instead of `Key` in `find(…)`.

Reply:
We have all these find() methods that use the ID type (Key in our case) because the CrudRepository's find methods are based on the ID type. I suppose we could also add find methods that take the entity.

Reply:
Just wanted to follow up that this `getId` is added as a convenience for the user. I imagined that the user would likely have a SpannerOperations on hand autowired and he could get the IDs of objects easily. For example, he might retrieve objects, then deleteById some of them. 

I see what you're saying about our find/findall methods taking in Object instead of Key. The reasoning at the moment is that we haven't planned to support any type other than Key, but I can imagine in the future that we check what type that Object actually is and either use it straight if it is a Key or call Key.of() if it's a primitive type etc...